lib/geoservices/queryable.rb

```diff
@@ -14,7 +14,15 @@ def query(layer_idx=0, options={})
     # returns nil if the layer does not have the Capability: Query
     def count(layer_idx=0, options={})
       return nil unless self.metadata["capabilities"] =~ /Query/
-      query(layer_idx, options.merge(:returnCountOnly => true))
+      # Current Version is not returned at lower than 10.1
+      unless self.metadata['currentVersion'].nil?
+        query(layer_idx, options.merge(:returnCountOnly => true))
+      else
+        # Return count only is not supported by versions lower than 10.1
+        # So request all the object ids and count them up
+        response = query(layer_idx, options.merge(:where => '1=1', :returnIdsOnly => true))
```

Comment:
Check out this request:
http://sampleserver3.arcgisonline.com/ArcGIS/rest/services/Petroleum/WellsWithFormations/MapServer/0/query?where=1=1&returnIdsOnly=true&f=json. Careful, it may crash your jsonview plugin it is so big :stuck_out_tongue: 

It takes a long time to cook, but it works.